not work "builtinGlobals" in rule "no-redeclare"

**Tell us about your environment**

* **ESLint Version:**
v4.0.0
* **Node Version:**
v8.0.0
* **npm Version:**
v5.0.0

Hi, maybe I finded bug.
Not wokr rule  property "builtinGlobals" in rule "no-redeclare".
```js
/*eslint no-redeclare: ["error", { "builtinGlobals": true }]*/
/*eslint-env browser*/

var top = 0;
var Object = 0;
```

console message without error "no-redeclare"

```bash
4:1   error  Unexpected var, use let or const instead        no-var
  4:5   error  'top' is assigned a value but never used        no-unused-vars
  4:11  error  Number constants declarations must use 'const'  no-magic-numbers
  5:1   error  Unexpected var, use let or const instead        no-var
  5:5   error  'Object' is assigned a value but never used     no-unused-vars
  5:14  error  Number constants declarations must use 'const'  no-magic-numbers
```

--- a/docs/rules/no-redeclare.md
+++ b/docs/rules/no-redeclare.md
@@ -32,6 +32,8 @@ If set to `true`, this rule also checks redeclaration of built-in globals, such 
 
 ### builtinGlobals
 
+The `"builtinGlobals"` option will check for redeclaration of built-in globals in global scope.
+
 Examples of **incorrect** code for the `{ "builtinGlobals": true }` option:
 
 ```js
@@ -50,3 +52,9 @@ var top = 0;
 ```
 
 The `browser` environment has many built-in global variables (for example, `top`). Some of built-in global variables cannot be redeclared.
+
+Note that when using the `node` or `commonjs` environments (or `ecmaFeatures.globalReturn`, if using the default parser), the top scope of a program is not actually the global scope, but rather a "module" scope. When this is the case, declaring a variable named after a builtin global is not a redeclaration, but rather a shadowing of the global variable. In that case, the `no-shadow` rule with the `"builtinGlobals"` option should be used.
+
+## Related Rules
+
+* [no-shadow](no-shadow.md)

--- a/docs/rules/no-redeclare.md
+++ b/docs/rules/no-redeclare.md
@@ -53,7 +53,7 @@ var top = 0;
 
 The `browser` environment has many built-in global variables (for example, `top`). Some of built-in global variables cannot be redeclared.
 
-Note that when using the `node` or `commonjs` environments (or `ecmaFeatures.globalReturn`, if using the default parser), the top scope of a program is not actually the global scope, but rather a "module" scope. When this is the case, declaring a variable named after a builtin global is not a redeclaration, but rather a shadowing of the global variable. In that case, the `no-shadow` rule with the `"builtinGlobals"` option should be used.
+Note that when using the `node` or `commonjs` environments (or `ecmaFeatures.globalReturn`, if using the default parser), the top scope of a program is not actually the global scope, but rather a "module" scope. When this is the case, declaring a variable named after a builtin global is not a redeclaration, but rather a shadowing of the global variable. In that case, the [`no-shadow`](no-shadow.md) rule with the `"builtinGlobals"` option should be used.
 
 ## Related Rules
 

